src/codec_dav1d.c

```diff
@@ -125,6 +126,23 @@ static avifBool dav1dCodecGetNextImage(struct avifCodec * codec,
         dav1d_data_unref(&dav1dData);
     }
 
+    // Drain all buffered frames in the decoder.
```

Comment:
I think it is better to document the purpose of the drain loop instead of focusing on the "single decoder instance" case. Even if we don't use a single codec instance to decode all tiles, for example in the case of an animated image file the call to decode the second frame will return the buffered frame from the earlier decode calls.

The fundamental issue here is that technically an AVIF "frame" can contain multiple OBUs and multiple AV1 frames underneath. While there are no practical uses for it, theoretically it is allowed. The purpose of this drain loop is to only return the first image in such cases and properly ignore the rest.

Either way, i agree that we need to document why we are supposedly draining the decoder after decoding every single frame. Because we generally associate the term drain/flush with an operation that is only performed once at the end. :)